dev/com.ibm.ws.artifact/src/com/ibm/wsspi/artifact/ArtifactEntry.java

```diff
@@ -79,12 +79,23 @@
      * <p>
      * This may return null if this Entry is a virtual container that does not have a real location on disk.
      * </p>
-     * 
+     *
      * @return The URL pointing to this entry on disk
      * @see ServletContext.getResource(String path)
      */
     URL getResource();
 
-    //Concepts absent, but under discussion. 
+    /**
+     * Returns the URL to use for the shared class cache to track if the class changes.
+     *
+     * For a jar it is the path of the jar. If it is another archive type, it is the patch to the archive with
+     * the entry, i.e. jar:file:<file>~/package/className.class. For a directory it is the directory URL.
+     *
+     * @param classFileResourceName class name in relative path form a/b/c.class
+     * @return URL to be used for the IBM / OpenJ9 shared class cache function.
```

Comment:
Like before - might want to mention that this could return null.

Reply:
As before the new method does mention it returning null.